File: pdu/python/std_msgs/pdu_cdr_conv_Header.py
from .pdu_pytype_Header import Header
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..builtin_interfaces.pdu_cdr_conv_Time import *



def py_to_cdr_body_Header(writer: CdrWriter, src: Header):
    py_to_cdr_body_Time(writer, src.stamp)
    writer.write_string(src.frame_id)


def cdr_body_to_py_Header(reader: CdrReader, dst: Header):
    cdr_body_to_py_Time(reader, dst.stamp)
    dst.frame_id = reader.read_string()
    return dst


def py_to_cdr_Header(src: Header) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Header(writer, src)
    return writer.bytes()


def cdr_to_py_Header(cdr_payload) -> Header:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Header()
    return cdr_body_to_py_Header(reader, dst)

File: pdu/javascript/std_msgs/pdu_cdr_conv_Header.js
import { Header } from './pdu_jstype_Header.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';


export class PduHeaderConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Header } src
     */
    static to_cdr_body(writer, src) {
        PduTimeConverter.to_cdr_body(writer, src.stamp);
        writer.write_string(src.frame_id);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Header } dst
     * @returns { Header }
     */
    static cdr_body_to_js(reader, dst) {
        PduTimeConverter.cdr_body_to_js(reader, dst.stamp);
        dst.frame_id = reader.read_string();
        return dst;
    }

    /**
     * @param { Header } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Header }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Header());
    }
}
